lib/membrane/rtp/session_bin.ex

```diff
@@ -588,4 +601,33 @@ defmodule Membrane.RTP.SessionBin do
     pt || PayloadFormat.get(encoding).payload_type ||
       raise "Cannot find default RTP payload type for encoding #{encoding}"
   end
+
+  defp maybe_handle_twcc(new_children, rtp_extensions, pad_ssrc, ctx, state) do
+    # workaround: as TWCC is a transport-wide extension, there should exist only one TWCC child
+    # that handles packets from all incoming streams that have declared support for it
+    {maybe_twcc, rtp_extensions} = Keyword.pop(rtp_extensions, :twcc)
+
+    should_link_twcc? = maybe_twcc != nil
+
+    {new_children, state} =
+      if should_link_twcc? and not Map.has_key?(ctx.children, :twcc) do
+        {twcc_ssrc, state} = add_ssrc(nil, state)
+
+        {Map.put(new_children, :twcc, %{maybe_twcc | sender_ssrc: twcc_ssrc}), state}
+      else
+        {new_children, state}
+      end
+
+    maybe_link_twcc =
+      if should_link_twcc? do
+        &(&1
+          |> via_in(Pad.ref(:input, pad_ssrc))
+          |> to(:twcc)
```

Comment:
Instead of doing the whole `new_children` and updating the state you could simply put that element in `to(:twcc, %{maybe_twcc | sender_ssrc: twcc_ssrc})` keeping the `new_children` out of this funciton, and just update the state respectively if you need to add the `ssrc`.

Reply:
Done